DeAceroDLabs/marketplace-frontend
# [DMP-740] Add Offer Context and Product detail endpoint

## What is done

[[DMP-740] Guardar Oferta Seleccionada](https://deacero.atlassian.net/browse/DMP-740)

## How is done

1. Added a new context for storing the selected offer.
2. Added a new route for getting the article details. (this needs to change as now it is made directly to Roobyn).
3. In the Offer context, added a param for storing the details of the articles that belong to the offer.

## How to test

1. Run the project via `npm start`
4. Add products to your cart.
5. Go to the /carrito page.
6. Click on "pagar"
7. You will see the available offers.
8. Select any of the offers.
9. Open the developer's console.
10. If you have the React extension added to your browser, you can navigate to the offers component and check for the state of the component.
11. Check that the values for selected offer and offerArticles change.
12. If you don't have the extension installed, you can make a console log inside the offers.tsx file to log the selectedOfferContext.

## Evidence

Add images or screen  recordings on how the feature was working after and before the fix (if applies)
